### DISCUSSION TITLE
Apache Error: 'Action start failed' with ParaView 5.12.0 in Docker Setup

### DISCUSSION THREAD

**Discussion by evdkmv** [Kitware/trame#522] [Q&A] [answered]
Hello, 

My docker setup works fine with ParaView 5.11.0

```

FROM --platform=linux/amd64 kitware/trame:py3.9
# FROM --platform=linux/amd64 kitware/trame:py3.10

RUN install -d -o trame-user -g trame-user /deploy
RUN apt update -y && apt install -y libosmesa6-dev

ARG PV_URL='https://www.paraview.org/files/v5.11/ParaView-5.11.0-osmesa-MPI-Linux-Python3.9-x86_64.tar.gz'
# ARG PV_URL='https://www.paraview.org/files/v5.12/ParaView-5.12.0-osmesa-MPI-Linux-Python3.10-x86_64.tar.gz'

RUN mkdir -p /opt/paraview && cd /opt/paraview && wget -qO- $PV_URL | tar --strip-components=1 -xzv
ENV TRAME_PARAVIEW=/opt/paraview

COPY --chown=trame-user:trame-user . /deploy

RUN /opt/trame/entrypoint.sh build
```

However, when I attempt to use `ParaView 5.12.0` with the base image `kitware/trame:py3.10`, I encounter the following error:

```
 * Restarting Apache httpd web server apache2                                                                                                 AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.2. Set the 'ServerName' directive globally to suppress this message
Action 'start' failed.
The Apache error log may have more information.
```

Unfortunately, when I check /deploy/server/logs/apache/error.log, there is nothing inside. Could anyone suggest where I should begin debugging this issue?

**Comment by jourdain** [comment]
Might be related to https://discourse.paraview.org/t/is-headless-mode-not-working-or-am-i-missing-something/14599/4

**Comment by jourdain** [comment]
Did you try another dummy trame app with trame:py3.10 to see if you run into similar issue?

**Reply by evdkmv** [reply]
So, when I use the `kitware/trame:py3.10` as a base image for the simple cone Paraview example, the same error appears. However, I can run the simple cone VTK example using `kitware/trame:py3.10` with no errors.

**Reply by jourdain** [reply]
It seems to be related to ParaView. Did you look at the session logs in `/deploy/server/logs/launcher/*.txt` to see if you get any error?

**Reply by evdkmv** [reply]
Unfortunately all logs in /deploy/server/logs/* (access.log, error.log, launcherLog.log) are empty. 

However I found other logs.

This is /var/log/apache2/error.log for py:3.9 and paraview 5.11

```
[Tue May 14 06:26:17.982248 2024] [mpm_event:notice] [pid 114:tid 274907613056] AH00489: Apache/2.4.41 (Ubuntu) configured -- resuming normal operations
[Tue May 14 06:26:17.986192 2024] [core:notice] [pid 114:tid 274907613056] AH00094: Command line: '/usr/sbin/apache2'
[Tue May 14 06:26:50.867034 2024] [core:warn] [pid 117:tid 274907613056] AH00098: pid file /var/run/apache2/apache2.pid overwritten -- Unclean shutdown of previous Apache run?
[Tue May 14 06:26:50.875065 2024] [mpm_event:notice] [pid 117:tid 274907613056] AH00489: Apache/2.4.41 (Ubuntu) configured -- resuming normal operations
[Tue May 14 06:26:50.877613 2024] [core:notice] [pid 117:tid 274907613056] AH00094: Command line: '/usr/sbin/apache2'
```

This is /var/log/apache2/error.log for py:3.10 and paraview 5.12 

```
[Tue May 14 06:37:29.678163 2024] [core:emerg] [pid 109:tid 274907801792] (95)Operation not supported: AH00023: Couldn't create the proxy mutex 
[Tue May 14 06:37:29.680291 2024] [proxy:crit] [pid 109:tid 274907801792] (95)Operation not supported: AH02478: failed to create proxy mutex
AH00016: Configuration Failed
```
following the discussion [here ](https://stackoverflow.com/questions/73294020/docker-couldnt-create-the-mpm-accept-mutex) I added `RUN echo "Mutex posixsem" >> /etc/apache2/apache2.conf`. Now apache starts, but I see an error in  `/deploy/server/logs/launcher/*`

```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
qemu: uncaught target signal 6 (Aborted) - core dumped
error: exception occurred: Subprocess aborted

```

I am on Mac M2.

**Reply by jourdain** [reply]
We don't have a Paraview for Linux Arm, hence the issue.

**Reply by jourdain** [reply]
Or at least that is all I can think of with some architecture/rosetta issue.

**Reply by evdkmv** [reply]
I thought the `--platform=linux/amd64` was the trick to solve the problem with ARM architecture. Apparently, it does not help with ParaView 5.12.

**Comment by jourdain** [comment, marked as answer]
Summary from discussion:
- Get the info you need from `/deploy/server/logs/launcher/*.txt`
- Might be [OSMesa issue with 5.12](https://discourse.paraview.org/t/is-headless-mode-not-working-or-am-i-missing-something/14599/4)

**Reply by evdkmv** [reply]
Thanks @jourdain for your help!
